modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/selector/GreaterOrEqual.java

```diff
@@ -0,0 +1,36 @@
+package io.ballerina.messaging.broker.core.selector;
+
+import io.ballerina.messaging.broker.core.Metadata;
+
+/**
+ * Implementation of a boolean expression. Here we compare two expressions and evaluate to a boolean value.
+ */
+
+public class GreaterOrEqual implements BooleanExpression {
+
+    private final Expression<Metadata> left;
+
+    private final Expression<Metadata> right;
+
+    public GreaterOrEqual (Expression<Metadata> left , Expression<Metadata> right) {
+        this.left = left;
+        this.right = right;
+    }
+
+    @Override
+    public boolean evaluate(Metadata metadata) {
+        Object leftValue = left.evaluate(metadata);
+        Object rightValue = right.evaluate(metadata);
+        if (leftValue == null || rightValue == null) {
+            return false;
+        }
+        String s = String.valueOf(leftValue);
+        String s1 = String.valueOf(rightValue);
+        double x = s.compareTo(s1);
```

Comment:
Here line 22 to 28 repeated in four other places. Is there any good way to implement a common function and use it?